### ISSUE TITLE
`consistent-boolean-name` incorrectly flagging destructured boolean variables that have a valid prefix

### ISSUE THREAD

**Issue by timdavish** [sindresorhus/eslint-plugin-unicorn#3483]
The `consistent-boolean-name` rule incorrectly flags destructured boolean variables that already have a valid prefix (e.g. `isX`) when the destructuring pattern has a default value assignment (`= {}`). Without the default value, the rule correctly recognizes the name as valid.

Rule: `consistent-boolean-name`

This is correctly accepted (no error):

```js
interface Options {
  isX?: boolean
}

const test = ({isX}: Options) => isX
```

This incorrectly triggers the rule:

```js
interface Options {
 isX?: boolean
}

const test = ({isX}: Options = {}) => isX
```

The only difference is the `= {}` default value for the parameter. The variable name `isX` already has a valid boolean prefix and should not be flagged in either case.
